packages/grid/src/vaadin-grid.js

```diff
@@ -979,6 +999,25 @@ class Grid extends ElementMixin(
     };
   }
 
+  /**
+   * @param {Event} event
+   * @protected
+   */
+  _showTooltip(event) {
+    // Only fire events if there is a slotted vaadin-tooltip element.
+    if (this._tooltipController.node && this._tooltipController.node.isConnected) {
+      const context = this.getEventContext(event);
+      this.dispatchEvent(new CustomEvent('tooltip-target-changed', { detail: { target: event.target } }));
+      this.dispatchEvent(new CustomEvent('tooltip-context-changed', { detail: { context } }));
+      this.dispatchEvent(new CustomEvent('tooltip-opened-changed', { detail: { opened: true } }));
+    }
```

Comment:
Since we have a direct reference to the controller, I guess we don't need the events after all.
```js
this._tooltipController.setTarget(event.target);
this._tooltipController.setContext(context);
this._tooltipController.setOpened(true);
```

Reply:
Done.